packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx

```diff
@@ -0,0 +1,442 @@
+import { Component, Element, h, Host, Prop, State, Watch } from "@stencil/core";
+import { IcPosition, IcSizes, IcThemeMode } from "../../utils/types";
+import { TableOfContentsItem } from "./ic-table-of-contents-item";
+import { DEVICE_SIZES } from "../../utils/helpers";
+
+const TOC_CONTENT_CLASS = "table-of-contents-content";
+const TOC_LIST_ITEMS_CLASS = "table-of-contents-list-items";
+const TOC_ITEM_CLASS = "table-of-contents-item-content";
+
+interface RankedElement extends Element {
+  trueIndex: number;
+  rank: string;
+  treeChildren: RankedElement[];
+}
+
+@Component({
+  tag: "ic-table-of-contents",
+  styleUrl: "ic-table-of-contents.css",
+  shadow: true,
+})
+export class TableOfContents {
+  @Element() el: HTMLIcTableOfContentsElement;
+
+  /**
+   * If `true`, the table of contents will display as black in the light theme, and white in dark theme.
+   */
+  @Prop() monochrome?: boolean = false;
+
+  /**
+   * Sets the theme color to the dark or light theme color. "inherit" will set the color based on the system settings or ic-theme component.
+   */
+  @Prop() theme?: IcThemeMode = "inherit";
+
+  /**
+   * If `true`, the table of contents item label will be truncated.
+   */
+  @Prop() truncate: boolean = true;
+
+  /**
+   * The size of the table of contents items.
+   */
+  @Prop() size: IcSizes = "medium";
+
+  /**
+   * Whether the table of contents will appear on the right of or above child content on large viewport widths.
+   */
+  @Prop() position: IcPosition = "right";
```

Comment:
i think this needs to be `mutable` as it complains about the value being changed in the unit tests